### PR TITLE
Have spanFrom work like root when the tracer's extract returns null

### PR DESCRIPTION
**Pull request by balajirrao**
When the tracer's extract method returned null, spanFrom was a no-op. Fix it
to work like root in this case.

### CODE REVIEW

**Comment by balajirrao** [comment]
if the new behaviour of `spanFrom` is acceptable, I'll add a test for it in this PR.

**Review by runtologist — COMMENTED** [review]
Thanks for this contribution. Sounds reasonable, please go ahead.

**Comment by balajirrao** [comment]
@runtologist, I've added the tests. Happy to hear your feedback!

**Comment by balajirrao** [comment]
@runtologist, gentle reminder to review this PR 😄

File: opentracing/src/main/scala/zio/telemetry/opentracing/OpenTracing.scala [hunk 1]
```diff
@@ -103,10 +103,9 @@ object OpenTracing {
...
-        .fold(_ => None, Option.apply)
-        .flatMap {
-          case None          => zio
-          case Some(spanCtx) =>
+        .foldM(
+          _ => zio,
+          spanCtx =>
             for {
               current <- service.currentSpan.get
               span    <- UIO(service.tracer.buildSpan(operation).asChildOf(spanCtx).start())
```
Comment (nsauro) [inline opentracing/src/main/scala/zio/telemetry/opentracing/OpenTracing.scala:111]: Please pardon my lurking, but I've been playing around with this library, and tracing in general. I too ran into this same problem, so this PR interested me. 

According to the documentation: https://github.com/opentracing/opentracing-java/blob/master/opentracing-api/src/main/java/io/opentracing/Tracer.java#L138-L152 if you pass in a `null` `SpanContext` or `Span`, it is a no op.

Do you think passing in a create function into the `spanFrom` would help?

Comment (balajirrao) [reply opentracing/src/main/scala/zio/telemetry/opentracing/OpenTracing.scala:111]: The `childOf` would be a no-op, which is what we indeed want. Without the PR, even `buildSpan` wouldn't be called if `spanCtx` was null. Does this help or did I not understand your question at all? 😄

**Review by runtologist — APPROVED** [review]
LGTM. Thank you and sorry this too so long.

File: opentracing/src/main/scala/zio/telemetry/opentracing/OpenTracing.scala (hunk 1 above)
Comment (nsauro) [reply opentracing/src/main/scala/zio/telemetry/opentracing/OpenTracing.scala:111]: Ah, looking at this with fresher eyes, I see the intention.  This was me just getting tripped up on ZIO code which I'm still learning.  Looks much better now!

**Comment by balajirrao** [comment]
@runtologist, no problem, thank you! I would have thought the next step would be to merge it - but I can't, due to lack of write access.

**Pull request merged**